src/transport/raw/PeerAddress.h

```diff
@@ -121,79 +217,41 @@ class PeerAddress
+        + 3                                      // " (+"
+        + 2                                      // integer for "more", low digits expected
+        + 6                                      // " more)"
         + 1;                                     // NullTerminator
 
+    static_assert(kMaxPeerDestinations < 100, "for maxStringSize to be correct for the 'more' part");
+
     template <size_t N>
     inline void ToString(char (&buf)[N]) const
     {
         ToString(buf, N);
     }
 
-    void ToString(char * buf, size_t bufSize) const
-    {
-        char ip_addr[Inet::IPAddress::kMaxStringLength];
-
-        char interface[Inet::InterfaceId::kMaxIfNameLength + 1] = {}; // +1 to prepend '%'
-        if (mInterface.IsPresent())
-        {
-            interface[0]   = '%';
-            interface[1]   = 0;
-            CHIP_ERROR err = mInterface.GetInterfaceName(interface + 1, sizeof(interface) - 1);
-            if (err != CHIP_NO_ERROR)
-            {
-                Platform::CopyString(interface, sizeof(interface), "%(err)");
-            }
-        }
-
-        switch (mTransportType)
-        {
-        case Type::kUndefined:
-            snprintf(buf, bufSize, "UNDEFINED");
-            break;
-        case Type::kUdp:
-            mIPAddress.ToString(ip_addr);
-#if INET_CONFIG_ENABLE_IPV4
-            if (mIPAddress.IsIPv4())
-                snprintf(buf, bufSize, "UDP:%s%s:%d", ip_addr, interface, mPort);
-            else
-#endif
-                snprintf(buf, bufSize, "UDP:[%s%s]:%d", ip_addr, interface, mPort);
-            break;
-        case Type::kTcp:
-            mIPAddress.ToString(ip_addr);
-#if INET_CONFIG_ENABLE_IPV4
-            if (mIPAddress.IsIPv4())
-                snprintf(buf, bufSize, "TCP:%s%s:%d", ip_addr, interface, mPort);
-            else
-#endif
-                snprintf(buf, bufSize, "TCP:[%s%s]:%d", ip_addr, interface, mPort);
-            break;
-        case Type::kBle:
-            // Note that BLE does not currently use any specific address.
-            snprintf(buf, bufSize, "BLE");
-            break;
-        default:
-            snprintf(buf, bufSize, "ERROR");
-            break;
-        }
-    }
+    void ToString(char * buf, size_t bufSize) const;
 
     /****** Factory methods for convenience ******/
 
-    static PeerAddress Uninitialized() { return PeerAddress(Inet::IPAddress::Any, Type::kUndefined); }
+    static PeerAddress Uninitialized() { return PeerAddress(Type::kUndefined); }
 
     static PeerAddress BLE() { return PeerAddress(Type::kBle); }
-    static PeerAddress UDP(const Inet::IPAddress & addr) { return PeerAddress(addr, Type::kUdp); }
-    static PeerAddress UDP(const Inet::IPAddress & addr, uint16_t port) { return UDP(addr).SetPort(port); }
-    static PeerAddress UDP(const Inet::IPAddress & addr, uint16_t port, Inet::InterfaceId interface)
+
+    static PeerAddress UDP(const Inet::IPAddress & addr, uint16_t port = CHIP_PORT,
```

Comment:
Please do not default to CHIP_PORT here

Reply:
Existing code was already defaulting to this. I plan to do small changes at once.
The reason I claim  existing code did this: code before had a UDP(addr) implementation that was using CHIP_PORT.